src/network/networkd-link.c

```diff
@@ -3166,6 +3207,13 @@ int link_update(Link *link, sd_netlink_message *m) {
                                 if (r < 0)
                                         return log_link_warning_errno(link, r, "Could not update DHCPv6 DUID: %m");
                         }
+
+                        if (link->radv) {
+                                r = sd_radv_set_mac(link->radv, &link->mac);
+
```

Comment:
not that it matters, but i'd always keep the function calls and the error checks for it together, and not place additional whitespace lines between them. i.e. this whitespace line should probably go